### PR TITLE
fix: dialog close on refresh

### PR DESCRIPTION
**Pull request by dguyen**
## Description

This is an example demo of the issue, looking for suggestions and discussions.

Currently any open dialogs nested into a table column will close when the page is focused due to us running `router.refresh()` whenever `onFocus` callback is triggered.

This appears to be caused by the columns prop getting replaced, which causes the columns to get rerendered.

There are a few ways to go about this, this commit is a dirty example which prevents this from happening.

Another option is to add a stableColumn prop, which we then assume never change so we use the initial value only.

## Summary by CodeRabbit

- **New Features**
    - Enhanced data tables across various components with optimized column definitions using `useMemo`, improving performance and maintainability.
    - Added tooltips for template types in the `TemplatesDataTable` for better user experience.

- **Bug Fixes**
    - Ensured consistent data structure in `AdminDocumentResults` even when no documents are found.

- **Documentation**
    - Updated type definitions for data table columns to improve clarity and type safety.

- **Chores**
    - Minor updates to line references in translation files for improved localization accuracy.

### CODE REVIEW

**Comment by Mythie** [comment]
Given we have functions and such in the column definition a hacky useState with a `JSON.stringify` comparison is likely to fail

**Comment by dguyen** [comment]
Yeah, any suggestions?

**Comment by Mythie** [comment]
Terrible type inference shenanigans to extract the column definition out of the render function :(

**Comment by Mythie** [comment]
I had a POC on a stash that's still fairly broken but i'll work on getting something together for us

File: packages/app-tests/e2e/teams/team-members.spec.ts [hunk 1]
```diff
@@ -32,8 +32,12 @@ test('[TEAMS]: update team member role', async ({ page }) => {
   await page.getByRole('combobox').click();
   await page.getByLabel('Manager').click();
   await page.getByRole('button', { name: 'Update' }).click();
+
+  // TODO: Remove me, but i don't care for now
+  await page.reload();
```
[the remark this answers is not in the document]
Comment (Mythie) [reply packages/app-tests/e2e/teams/team-members.spec.ts:37]: please do

**Pull request merged**